tests/test_apply.py

```diff
@@ -42,132 +42,110 @@
+    def test_two_qubit_operations_no_parameters(self, init_state, device, operation, tol):
+        """Test that two qubit operations that take no parameters work fine
+        with the apply method."""
         dev = device(2)
         state = init_state(2)
+        wires = [0, 1]
 
-        dev.apply("QubitStateVector", [0, 1], [state])
-        dev.apply(name, [0, 1], [])
-        dev._obs_queue = []
-        dev.pre_measure()
+        applied_operation = operation(wires=wires)
+
+        dev.apply([qml.QubitStateVector(state, wires=wires), applied_operation])
 
         res = np.abs(dev.state) ** 2
-        expected = np.abs(mat @ state) ** 2
+        expected = np.abs(applied_operation.matrix @ state) ** 2
         assert np.allclose(res, expected, **tol)
 
-    @pytest.mark.parametrize("mat", [U, U2])
-    def test_qubit_unitary(self, init_state, device, mat, tol):
-        N = int(np.log2(len(mat)))
-        dev = device(N)
-        state = init_state(N)
+    @pytest.mark.parametrize("theta", [0.5432, -0.232])
+    @pytest.mark.parametrize("operation", two_qubit_param)
+    def test_two_qubit_operations_parameters(self, init_state, device, operation, theta, tol):
+        """Test that two qubit parametrized operations work fine with the
+        apply method."""
+        dev = device(2)
+        state = init_state(2)
+        wires = [0, 1]
+        applied_operation = operation(theta, wires=wires)
 
-        dev.apply("QubitStateVector", list(range(N)), [state])
-        dev.apply("QubitUnitary", list(range(N)), [mat])
-        dev._obs_queue = []
-        dev.pre_measure()
+        dev.apply([qml.QubitStateVector(state, wires=wires), applied_operation])
 
         res = np.abs(dev.state) ** 2
-        expected = np.abs(mat @ state) ** 2
+        expected = np.abs(applied_operation.matrix @ state) ** 2
         assert np.allclose(res, expected, **tol)
 
-    @pytest.mark.parametrize("name, mat", three_qubit)
-    def test_three_qubit_no_parameters(self, init_state, device, name, mat, tol):
+    @pytest.mark.parametrize("operation", three_qubit)
+    def test_three_qubit_operations_no_parameters(self, init_state, device, operation, tol):
+        """Test that three qubit operations that take no parameters work fine
+        with the apply method."""
         dev = device(3)
         state = init_state(3)
+        applied_operation = operation(wires=[0, 1, 2])
 
-        dev.apply("QubitStateVector", [0, 1, 2], [state])
-        dev.apply("QubitUnitary", [0, 1, 2], [mat])
-        dev._obs_queue = []
-        dev.pre_measure()
-
-        res = np.abs(dev.state) ** 2
-        expected = np.abs(mat @ state) ** 2
-        assert np.allclose(res, expected, **tol)
-
-    @pytest.mark.parametrize("theta", [0.5432, -0.232])
-    @pytest.mark.parametrize("name,func", two_qubit_param)
-    def test_single_qubit_parameters(self, init_state, device, name, func, theta, tol):
-        """Test PauliX application"""
```

Comment:
It was probably not related to this PR, but a few tests still have the docstring   """Test PauliX application""". Maybe a good opportunity to correct?

Reply:
Thanks! Could you check if these are alright with the latest commit? I've pushed one yesterday after noticing this and that should clear these up

Reply:
Great, looks good!